Lib/test/test_capi/test_abstract.py

```diff
@@ -1007,6 +1007,46 @@ def test_object_generichash(self):
         for obj in object(), 1, 'string', []:
             self.assertEqual(generichash(obj), object.__hash__(obj))
 
+    def run_iter_api_test(self, next_func):
+        for data in (), [], (1, 2, 3), [1 , 2, 3], "123":
+            with self.subTest(data=data):
+                items = []
+                it = iter(data)
+                while (item := next_func(it)) is not None:
+                    items.append(item)
+                self.assertEqual(items, list(data))
+
+        class Broken:
+            def __init__(self):
+                self.count = 0
+
+            def __next__(self):
+                if self.count < 3:
+                    self.count += 1
+                    return self.count
+                else:
+                    raise TypeError('bad type')
+
+        it = Broken()
+        self.assertEqual(next_func(it), 1)
+        self.assertEqual(next_func(it), 2)
+        self.assertEqual(next_func(it), 3)
+        with self.assertRaisesRegex(TypeError, 'bad type'):
+            next_func(it)
+
+    def test_iter_next(self):
+        from _testcapi import PyIter_Next
+        self.run_iter_api_test(PyIter_Next)
+        # CRASHES PyIter_Next(10)
```

Comment:
It took me a while to understand that the test is not about `PyIter_Next(10)`. Maybe say something like:

```suggestion
		self.run_iter_api_test(PyIter_Next)
		# PyIter_Next(10) cannot be tested since PyIter_Next()
		# expects its argument to be an iterator, yet it does
		# not check it explicitly nor raises an exception.
```

Reply:
It's a convention: `git grep CRASHES Lib/test`. Calling `PyIter_Next(10)` will segfault.

Reply:
Oh thanks for the tip!